compiler/semantic/scope.go

```diff
@@ -153,6 +159,19 @@ func (s *Scope) resolve(n ast.Node, path field.Path) (sem.Expr, error) {
 	return sem.NewThis(n, path), nil
 }
 
+func (s *Scope) indexBase() int {
+	scope := s
+	for {
+		if v, ok := s.pragmas["index_base"]; ok {
+			return v.(int)
+		}
+		if scope.parent == nil {
+			return 0
+		}
+		scope = scope.parent
+	}
```

Comment:
This doesn't seem right since it's equivalent this:

```go
	if v, ok := s.pragmas["index_base"]; ok {
		return v.(int)
	}
	return 0
```

Reply:
How the heck are the test passing tho